packages/salesforce/src/Adaptor.js

```diff
@@ -148,16 +148,19 @@ export function retrieve(sObject, id, callback) {
  * Execute an SOQL query.
  * Note that in an event of a query error,
  * error logs will be printed but the operation will not throw the error.
+ *
+ * The Salesforce query API is subject to rate limits, {@link https://sforce.co/3W9zyaQ See for more details}.
  * @public
  * @example
  * query(state=> `SELECT Id FROM Patient__c WHERE Health_ID__c = '${state.data.field1}'`);
  * @example <caption>Query more records if next records are available</caption>
  * query(state=> `SELECT Id FROM Patient__c WHERE Health_ID__c = '${state.data.field1}'`, { autoFetch: true });
  * @function
- * @param {String} qs - A query string.
+ * @param {String} qs - A query string. Must be less than `4000` characters in WHERE clause
  * @param {Object} options - Options passed to the bulk api.
- * @param {boolean} [options.autoFetch] - Fetch next records if available.
+ * @param {boolean} [options.autoFetch=false] - Fetch next records if available.
  * @param {Function} callback - A callback to execute once the record is retrieved
+ * @throws {Error} Throws an error if characters in SOQL WHERE clauses exceed `4,000` characters
```

Comment:
This is fine but there  are other limits, right? So I'm a bit concerned about calling this one out in particular

For example the SOQL statement in total can only be 100k characters. We're not talking about that here.

I'm not even sure it's accurate to say `throws an error of rate limits are exceeded`, because I guess it'll also throw if unauthorised, or the query syntax is invalid, or the query is null.

I still  think we've done enough by adding a link to the limits.  By mentioning some limits explicitly, and not others, we're only adding confusion.

If you want to help developers, you need to catch errors in the adaptor and include useful error messages, like. Like catching a `bad message 414` and linking to the rate limit docs.

Reply:
Okay got you, i have opened up an issue to track this suggestion https://github.com/OpenFn/adaptors/issues/531